html/obsolete/requirements-for-implementations/other-elements-attributes-and-apis/document-color-fg.html

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<title>document wg color</title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/obsolete.html#dom-document-fgcolor">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/obsolete.html#dom-document-linkcolor">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/obsolete.html#dom-document-vlinkcolor">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/obsolete.html#dom-document-alinkcolor">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/obsolete.html#dom-document-bgcolor">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/obsolete.html">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body text="red" linkColor="green" vlink="blue" alink="orange" bgColor="yellow">
+<script>
+test(function() {
+  assert_equals(document.fgColor, document.body.text);
+
+  assert_equals(document.linkColor, document.body.link);
+
+  assert_equals(document.vlinkColor, document.body.vLink);
+
+  assert_equals(document.alinkColor, document.body.aLink);
+
+  assert_equals(document.bgColor, document.body.bgColor);
+})
+test(function() {
+  document.body.text = red;
+
+  assert_equals(document.fgColor, "red");
+}, "document's fgColor IDL attribute reflects the body's text content attribute");
+
+test(function() {
+  document.body.link = green;
+
+  assert_equals(document.linkColor, "green");
+}, "document's linkColor IDL attribute reflects the body's link content attribute");
+
+test(function() {
+  document.body.vLink = blue;
+
+  assert_equals(document.vlinkColor, "blue");
+}, "document's vlinkColor IDL attribute reflects the body's vLink content attribute");
+
+test(function() {
+  document.body.aLink = orange;
+
+  assert_equals(document.alinkColor, "orange");
+}, "document's alinkColor IDL attribute reflects the body's aLink content attribute");
+
+test(function() {
+  document.body.bgColor = yellow;
+
+  assert_equals(document.bgColor, "yellow");
+}, "document's bgColor IDL attribute reflects the body's bgColor content attribute");
+</script>
```

Comment:
I think we need one more set of tests, where we finally remove the attributes from `body` to make sure that they get reflected properly in `document`. So:

```JS
test(function(){
   document.body.removeAttribute("test");
   assert_equals(document.fgColor, "");
   // and so on...
}, "Removing the attributes causes them to be reflected as the empty string");
```

Reply:
Aren't all these reflection tests already done in an automated fashion by https://github.com/web-platform-tests/wpt/blob/master/html/dom/elements-sections.js ?

Reply:
Indeed... we don't need these tests :) @akansha2608 was learning how to read/process spec text. 

So, I think we should abandon this and do the "TODO" in https://github.com/web-platform-tests/wpt/blob/master/html/dom/elements-sections.js

Reply:
So yeah, looking at the above, we could basically run everything above in a frameset and on a new body. 

@akansha2608, ping me on Slack on Monday and I’ll help you with it. In the meantime, read over the original bug to get a more clear sense of exactly what we need to test.